### PR TITLE
feat: xcm exchange asset

### PR DESCRIPTION
**Pull request by dmoka**
## Description

We enable AssetExchange xcm message instruction in hydra.

It also means that we needed to change the `XcmFilter` of pallet_xcm from `Nothing` to `AllowTransferAndSwap`, which is a more permissive in terms of allowed xcm instructions.

## Checklist:

- [ ] I have updated the documentation if necessary.
- [ ] I have added tests to cover my changes, regression test if fixing an issue.
- [ ] This is a breaking change.

### CODE REVIEW

File: runtime/hydradx/src/xcm.rs [hunk 1]
```diff
@@ -210,7 +216,7 @@ impl pallet_xcm::Config for Runtime {
 	type SendXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
 	type XcmRouter = XcmRouter;
 	type ExecuteXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
-	type XcmExecuteFilter = Nothing;
+	type XcmExecuteFilter = AllowTransferAndSwap<MaxXcmDepth, MaxNumberOfInstructions, RuntimeCall>;
```
Comment (mrq1911) [inline runtime/hydradx/src/xcm.rs:219]: why is this change necessary? i dont think we are planning to execute xcm messages locally and xcm::execute extrinsic is call filtered anyway currently

**Review by mrq1911 — CHANGES_REQUESTED** [review]
did you try to exchange something from some sibling parachain using chopstics?

File: runtime/hydradx/src/xcm.rs (hunk 1 above)
Comment (apopiak) [reply runtime/hydradx/src/xcm.rs:219]: We want it in combination with changing the call filter to allow UIs to interact with the chain through `pallet_xcm::execute`. This is part of the XCM exchange asset stuff.
Concretely you could imagine our UI swapping tokens on Hydra and then sending them to the destination in one extrinsic/with one XCM.

**Review by jak-pan — COMMENTED** [review]
Any problems with this?
